feat(aio): about page

sjtrimble opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)



**What is the new behavior?**



**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

sjtrimble committed:
feat(aio): about page styles

contributor fix

contributor card styling updates

button styling and container size fix

remove button and add image border

selection filter for buttons

header only show when group selected and styles

initial flip card funcitonality and data change

selected buttons styling

complete flip func and card style

flip feature

contributor group changes

fixes

mary-poppins commented:
The angular.io preview for cabf32c is available [here][1].

[1]: https://pr15914-cabf32cea53ae99235aaed4412e249815d33731f.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 0d1f773 is available [here][1].

[1]: https://pr15914-0d1f773db196a879d3e798f56271de600ec9737e.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 432391a is available [here][1].

[1]: https://pr15914-432391ae2ce6a80807fa9078b1838c99de74faea.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 8e95783 is available [here][1].

[1]: https://pr15914-8e95783375a307cdfe55f252a633e642c4f9fa7b.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 1f4abfe is available [here][1].

[1]: https://pr15914-1f4abfe98c4f07ed5293d20a0e92bcd17532545c.ngbuilds.io/

petebacondarwin commented on aio/content/marketing/index.html:
```diff
@@ -66,8 +66,6 @@ <h1 class="text-headline">One framework.<br>Mobile &amp; desktop.</h1><a
-    <div class="cta-bar"><a href="/docs/ts/latest/quickstart.html" md-button="md-button"
-                            class="button button-large button-shield mat-raised mat-primary">Get
-      Started</a></div>
+    <div class="cta-bar"><a href="/docs/ts/latest/quickstart.html" md-button="md-button" class="button button-large button-shield mat-raised mat-primary">Get Started</a></div>
```
The URL here is incorrect. I think it should be simply: `guide/quickstart`. For example: https://aio-staging.firebaseapp.com/guide/quickstart

petebacondarwin commented on aio/src/app/embedded/contributor/contributor-list.component.ts:
```diff
@@ -5,21 +5,34 @@ import { ContributorService } from './contributor.service';
+  showContributorGroup(group) {
+    console.log('Group: ', group, 'selectedGroup: ', this.selectedGroup);
+    this.selectedGroup = group;
+    console.log('selectedGroup: ', this.selectedGroup);
```
Can we remove these console logs now that you are happy that it works?

petebacondarwin commented on aio/src/app/embedded/contributor/contributor.component.ts:
```diff
@@ -2,35 +2,54 @@ import { Component, Input } from '@angular/core';
+    <div [ngClass]="{ 'flipped': person.isFlipped }" class="contributor-card">
+    
+        <div class="card-front" (click)="flipCard(person)">
+            <div *ngIf="person.picture" class="contributor-image" [style.background-image]="'url('+pictureBase+person.picture+')'">
```
Why use a background image rather than an image tag?
Doesn't this impact accessibility?

sjtrimble replied:
In order to get a centered round image (from a square or rectangular image of any size).

petebacondarwin commented on aio/src/app/embedded/contributor/contributor.component.ts:
```diff
@@ -2,35 +2,54 @@ import { Component, Input } from '@angular/core';
+    pictureBase = 'content/images/bios/';
+    
+    flipCard(person){
+        console.log('Flipped Status: ', this.person.isFlipped);
```
Let's remove the logging. It must only have been used to debug during development.

petebacondarwin commented on aio/src/app/embedded/contributor/contributor.component.ts:
```diff
@@ -2,35 +2,54 @@ import { Component, Input } from '@angular/core';
 
 import { Contributor } from './contributors.model';
 
+import { DomSanitizer, SafeResourceUrl, SafeUrl} from '@angular/platform-browser';
```
these don't appear to be used

petebacondarwin commented on aio/src/app/embedded/contributor/contributor.component.ts:
```diff
@@ -2,35 +2,54 @@ import { Component, Input } from '@angular/core';
 import { Contributor } from './contributors.model';
 
+import { DomSanitizer, SafeResourceUrl, SafeUrl} from '@angular/platform-browser';
+import { Pipe, PipeTransform } from '@angular/core';
```
these don't appear to be used either

petebacondarwin reviewed (dismissed):
Along with my other comments, the CircleCI test is failing due to invalid commit messages and the Travis failures appear to be due to linting errors in the code.

Perhaps it would be best to just squash all the commits?
And you can check the linting by running `yarn lint` I think.

sjtrimble commented:
@petebacondarwin I squashed the commits and cleaned up the code.

petebacondarwin commented:
Sorry @sjtrimble - it still fails `yarn lint`: https://travis-ci.org/angular/angular/jobs/222292909#L972

gkalpak commented on aio/content/marketing/contributors.json:
```diff
@@ -5,7 +5,7 @@
     "website": "http://misko.hevery.com",
     "bio": "Miško Hevery is the creator of AngularJS framework. He has passion for making complex things simple. He currently works at Google, but has previously worked at Adobe, Sun Microsystems, Intel, and Xerox, where he became an expert in building web applications in web related technologies such as Java, JavaScript, Flex and ActionScript.",
-    "group": "Lead"
+    "group": "Angular"
```
OOC, where did these changes come from?

sjtrimble replied:
Jules requested we change the groups to Angular, GDE and Community. Something new :) We'll need to add all of the GDEs as most are not currently there.

petebacondarwin commented:
@sjtrimble - also this conflicts in a big way with @wardbell's recent PR which we merged this weekend. Can you work with him to sort this out. For instance, he is sorting the contributor groups alphabetically, rather than by a hard coded order.

petebacondarwin committed:
fix(aio): fix linting issues

sjtrimble commented:
@petebacondarwin @wardbell This functionality is what was discussed with Jules which should remain. The alphabetical order should only apply within a selected group (i.e. Angular team members can be alphabetized).

wardbell committed:
feat(aio): about/contributor page rebased and tweaked

wardbell commented:
@sjtrimble I rebased and merged it with the changes I had made to the contributors before you got started. Should be good to go.

One thing in your department: the bio card chops off the bottom of the name. And I don't love centered for anything but very short text. Are you sure about that?
![image](https://cloud.githubusercontent.com/assets/129061/25075220/3c5c4534-22c4-11e7-836c-510ceb245cce.png)

wardbell reviewed (changes requested):
See my concerns about formatting of the bio card.
Otherwise LGTM

mary-poppins commented:
The angular.io preview for 0a54f16b4ebc7d359bc988b62aa368cc1f8f2181 is available [here][1].

[1]: https://pr15914-0a54f16b4ebc7d359bc988b62aa368cc1f8f2181.ngbuilds.io/

sjtrimble reviewed (commented):
We have links changed here that should remain as google.com, no?

petebacondarwin commented on aio/content/marketing/index.html:
```diff
@@ -66,8 +66,12 @@ <h1 class="text-headline">One framework.<br>Mobile &amp; desktop.</h1><a
       Started</a></div>
+=======
+    <div class="cta-bar"><a href="/guide/quickstart" md-button="md-button" class="button button-large button-shield mat-raised mat-primary">Get Started</a></div>
+>>>>>>> feat(aio): about page styles
```
There is an unresolved conflict here.

petebacondarwin commented on aio/src/app/embedded/contributor/contributor.service.spec.ts:
```diff
@@ -74,29 +74,37 @@ describe('ContributorService', () => {
-      "website": "https://google.com/+IgorMinar",
-      "bio": "Igor is a software engineer at Google.",
-      "group": "Lead"
+      "website": "https://Angular.com/+IgorMinar",
```
this doesn't affect the app, because it is only test data but, yes, this should be google not Angular

petebacondarwin replied:
And similarly below

wardbell replied:
LOL about the test data. My search and replace of "google" by "Angular" was a bit aggressive.

petebacondarwin reviewed (dismissed):
Fix the merge conflict

sjtrimble committed:
fix(aio): about page

Header now visible fully when card if flipped to back info

Contributor website links fixed

mary-poppins commented:
The angular.io preview for 147434645f1b446ae8a88cd3875d675fbcd74a35 is available [here][1].

[1]: https://pr15914-147434645f1b446ae8a88cd3875d675fbcd74a35.ngbuilds.io/
